src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftBuilder.java

```diff
@@ -158,9 +151,36 @@ public void onFailure(IOException e) {
                     }
                 }, null);
             }
+        }
+        return stop;
+    }
+
+    default void waitOnBuild(IClient client, long startTime, String bldId, TaskListener listener, Map<String, String> overrides, long wait, boolean follow, boolean chatty) throws InterruptedException {
+        IBuild bld = null;
+        String bldState = null;
+
+        // get internal OS Java REST Client error if access pod logs while bld is in Pending state
+        // instead of Running, Complete, or Failed
+
+        IStoppable stop = null;
+
+        final AtomicBoolean needToFollow = new AtomicBoolean(follow);
+
+        while (System.currentTimeMillis() < (startTime + wait)) {
+            bld = client.get(ResourceKind.BUILD, bldId, getNamespace(overrides));
+            bldState = bld.getStatus();
+            if (Boolean.parseBoolean(getVerbose(overrides)))
+                listener.getLogger().println("\nOpenShiftBuilder bld state:  " + bldState);
+
+            if (needToFollow.compareAndSet(true, false)) {
+                stop = this.getBuildPodLogs(client, bldId, overrides, chatty, listener, needToFollow);
```

Comment:
Nope, the second call only occurs if the first did not find the pod (the `stop == null` check)